File: 36-valid-sudoku/valid-sudoku.cpp
class Solution {
public:
    
    bool isValidSudoku(vector<vector<char>>& board) {
        bool row[9][9] = {false};
        bool col[9][9] = {false};
        bool sub[9][9] = {false};

        for(int r=0;r<9;r++){
            for(int c=0;c<9;c++){
                if(board[r][c]=='.')
                    continue;

                int idx= board[r][c]-'0'-1;
                int area=(r/3)*3+(c/3);

                 if(row[r][idx] || col[c][idx] || sub[area][idx]){
                    return false;
                }
                row[r][idx] = true;
                col[c][idx] = true;
                sub[area][idx] = true;
            }
            
        }
        return true;
    }
};

File: 36-valid-sudoku/valid-sudoku.py
class Solution:
    def isValidSudoku(self, board: List[List[str]]) -> bool:
        row = [[False] * 9 for _ in range(9)]
        col = [[False] * 9 for _ in range(9)]
        sub = [[False] * 9 for _ in range(9)]

        for r in range(len(board)):
            for c in range(len(board[0])):
                if(board[r][c] == '.'):
                    continue
                idx = int(board[r][c])-1
                area = (r//3)*3 + (c//3)

                if(row[r][idx] or col[c][idx] or sub[area][idx]):
                    return False
                
                row[r][idx] = True
                col[c][idx] = True
                sub[area][idx] = True

        return True

